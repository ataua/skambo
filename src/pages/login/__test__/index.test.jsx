import React from 'react';
import Login from '../index';
import TestRenderer from 'react-test-renderer';

jest.mock('react-redux', () => ({
  __esModule: true,
  useDispatch: () => jest.fn(),
}));

jest.mock('react-router-dom', () => ({
  __esModule: true,
  useHistory: jest.fn(),
  Link: () => <a></a>
}));

jest.mock('react-hook-form', () => ({
  __esModule: true,
  useForm: () => ({ register: jest.fn(), handleSubmit: jest.fn(), errors: jest.fn() }),
}));

test('Render Form component', () => {
  const tree = TestRenderer.create(<Login />).toJSON();

  expect(tree).toMatchSnapshot();
});

