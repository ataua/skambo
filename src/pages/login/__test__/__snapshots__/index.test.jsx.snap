// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Form component 1`] = `
<div
  className="sc-bdnylx leTbsp"
>
  <div
    className="sc-gtssRu brhlVN"
  >
    <div
      className="sc-hKFyIo dJqfuM"
    >
      <img
        alt="Logotipo Skambo"
        src="register-icon.svg"
      />
    </div>
    <div
      className="sc-dlnjPT cOIIvD"
    >
      <img
        className="sc-gKAblj kwtwHG"
        src="logotipo.png"
      />
      <form
        className="ui form"
        onSubmit={[Function]}
      >
        <div
          className="required field"
        >
          <label>
            <div>
              E-Mail
            </div>
            <input
              name="email"
              placeholder="Digite seu E-Mail"
            />
          </label>
        </div>
        <div
          className="required field"
        >
          <label>
            <div>
              Senha
            </div>
            <input
              name="password"
              placeholder="Digite sua Senha"
              type="password"
            />
          </label>
        </div>
        <button
          className="sc-eCApGN bqMDhY"
          type="submit"
        >
          Fazer Login
        </button>
      </form>
      <a />
      <span
        className="sc-iCoHVE ccpwEu"
      >
        
      </span>
    </div>
  </div>
</div>
`;
